pkg/volume/util.go

```diff
@@ -81,20 +81,46 @@ func internalRecycleVolumeByWatchingPodUntilCompletion(pvName string, pod *v1.Po
 	_, err = recyclerClient.CreatePod(pod)
 	if err != nil {
 		if errors.IsAlreadyExists(err) {
-			glog.V(5).Infof("old recycler pod %q found for volume", pod.Name)
+			deleteErr := recyclerClient.DeletePod(pod.Name, pod.Namespace)
+			if deleteErr != nil {
+				glog.V(5).Infof("failed to delete old recycler %s/%s: %s", pod.Namespace, pod.Name, deleteErr)
```

Comment:
Don't log if you're returning an error since that almost always results in it being double-logged. The exception is if the error is user-facing and you want to log additional details (add a comment if that's the case, and log at Error level).